### PR TITLE
Add infinite scroll feature

### PR DESCRIPTION
**Pull request by julienbourdeau**
If you turn the option on, you'll see a _load more_ button instead of the traditional pagination.

 - [x] Add configuration
 - [x] Add frontend feature
 - [x] Dig deeper url behavior

Any comments are welcome while I'm adding tests.

### Backend

configuration___system___magento_admin

### Frontend

men

### CODE REVIEW

File: app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml [hunk 1]
```diff
@@ -193,6 +193,9 @@ $autocompleteApiKey = $algoliaHelper->generateSearchSecuredApiKey($config->getSe
 $algoliaJsConfig = array(
     'instant' => array(
         'enabled' => $config->isInstantEnabled(),
+        'infiniteScroll' => array(
+                'enabled' => $config->isInfiniteScrollEnabled(),
```
Comment (JanPetr) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: Does it have to be nested array? How about just `enabledInfiniteScroll`?

File: app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml [hunk 2]
```diff
@@ -271,6 +274,7 @@ $algoliaJsConfig = array(
         'categories' => $this->__('Categories'),
         'products' => $this->__('Products'),
         'searchBy' => $this->__('Search by'),
+        'showMore' => $this->__('More products'),
```
Comment (JanPetr) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: I'd go for more descriptive "Show more products" than just "More products"

File: js/algoliasearch/instantsearch.js [hunk 3]
```diff
@@ -159,27 +159,44 @@ document.addEventListener("DOMContentLoaded", function (event) {
...
-			})
-		);
+				})
+			);
+		} else {
+			search.addWidget(
+				algoliaBundle.instantsearch.widgets.hits({
+					container: '#instant-search-results-container',
+					templates: {
+						allItems: '{{#hits}}' + $('#instant-hit-template').html() + '{{/hits}}'
```
Comment (JanPetr) [inline js/algoliasearch/instantsearch.js]: That we cannot do, people need to have option to update template without touching JS code.
And it's a huge BC break as current users may have custom template and this code would make it not work at all.

**Comment by redox** [comment]
That's so cool @JanPetr @julienbourdeau 👍  Does it still store the `page` query parameter in the URL? What's the impact when you share the URL with someone while being already on page 3?

File: app/design/frontend/base/default/template/algoliasearch/instantsearch/hit.phtml [hunk 4]
```diff
@@ -14,61 +14,61 @@ $price_key = $config->isCustomerGroupsEnabled(Mage::app()->getStore()->getStoreI
 
 <!-- Product hit template -->
 <script type="text/template" id="instant-hit-template">
-    {{#hits}}
```
Comment (JanPetr) [inline app/design/frontend/base/default/template/algoliasearch/instantsearch/hit.phtml]: Huge BC break, I wouldn't modify this template. I'd rather create a new template like `infinite-scroll-hit.phtml` and copy it.
It'll be harder for the maintenance, but better for BC POV, what do you think?

Or, you can create a template `simple-hit.phtml` and include it here in `hit.phtml`, so we would have code written only once with preserving BC. Hmm?

**Comment by julienbourdeau** [comment]
@redox So the page parameter is ignored but it seems like other parameters are well. I'll have a deeper look. Thanks!

File: app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml (hunk 1 above)
Comment (julienbourdeau) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: I had translations nested here before but I moved them. You're right, I'll remove one level.

**Comment by julienbourdeau** [comment]
Ok so the page parameters is now removed from the URL when infinite scroll is enabled.

Thank you all for your comments! 👍

**Comment by tataragne** [comment]
Hi, nice improvment ! Can you rebase this branch on 1.9.0 version ?

**Comment by julienbourdeau** [comment]
@tataragne I just rebased it on top of the develop branch 👍

**Comment by tataragne** [comment]
@julienbourdeau ty, @JanPetr any chance to get this in the next release ?

**Comment by JanPetr** [comment]
@julienbourdeau I tested it and everything looks OK! Just one small thing - when I go to the end, there is still "Show more products" button displayed. Can you hide it on the last page?

**Comment by julienbourdeau** [comment]
Oh yeah, I'll look into it!

**Comment by julienbourdeau** [comment]
I added it. First, I starter something overly complicated with a js eventlistener but then I found a much better solution. ^^

**Comment by magegirl** [comment]
Hello,
Is it possible that the scroll automatically starts without button "show more" ?

**Pull request merged**